src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+       col_grass = ispval
+       do il = 1, max_lunit
+          l = grc%landunit_indices(il, g)
+          if (l == ispval) cycle
+          if (lun%itype(l) == istsoil) then
+             do p = lun%patchi(l), lun%patchf(l)
+                is_grass = patch%itype(p) == nc4_grass &
+                     .or. patch%itype(p) == nc3_nonarctic_grass &
+                     .or. patch%itype(p) == nc3_arctic_grass
+                if (is_grass .and. col%wtgcell(patch%column(p)) > 1e-6) then
+                   col_grass = patch%column(p)
+                   exit
+                end if
+             end do
+          end if
+          if (col_grass /= ispval) exit
+       end do
+       ! Transfer of manure from all crop columns to the natural vegetation column:
+       flux_grass_graze = 0_r8
+       flux_grass_spread = 0_r8
+       flux_grass_spread_tan = 0_r8
+
+       do il = 1, max_lunit
+          l = grc%landunit_indices(il, g)
+          if (l == ispval) cycle
+          if (lun%itype(l) == istcrop) then
+             ! flux_avail = manure excreted per m2 of crops (ndep_mixed_grc = per m2 / all land units)
+             do c = lun%coli(l), lun%colf(l)
+                if (.not. col%active(c)) cycle
+                if (col%wtgcell(c) < 1e-6) cycle
+
+                if (col%landunit(c) /= l) then
+                   write(iulog, *) g, il, c, col%landunit(c)
+                   call endrun('something wrong')
+                end if
+                if (.not. any(c==filter_soilc(1:num_soilc))) then
+                   write(iulog, *) c, n_manure_spread_col(c)
+                   call endrun('column not in soilfilter')
+                end if
+
+                if (crop_man_is4crop_area) then
+                   invscale = 1.0_r8
+                else
+                   invscale = 1.0_r8 / lun%wtgcell(l)
+                end if
+
+                n_manure_mixed_col(c) = (ndep_ngrz_grc(g) + ndep_sgrz_grc(g)) * kg_to_g * invscale
+
+                tempr_min_10day = temperature_inst%t_a10min_patch(col%patchi(c))
+                if (tempr_min_10day > tempr_min_grazing) then
+                   ! fraction of animals grazing -> allocate some manure to grasslands before barns
+                   flux_grazing = max_grazing_fract * ndep_sgrz_grc(g) * kg_to_g * invscale
+                   flux_avail_rum = (ndep_sgrz_grc(g)*(1.0_r8 - max_grazing_fract)) * kg_to_g * invscale
+                   grz_fract(c) = max_grazing_fract
+                else
+                   flux_grazing = 0.0_r8
+                   flux_avail_rum = ndep_sgrz_grc(g) * kg_to_g * invscale
+                   grz_fract(c) = 0.0_r8
+                end if
+                flux_avail_mg = ndep_ngrz_grc(g) * kg_to_g * invscale
+                flux_grass_graze = flux_grass_graze + flux_grazing*col%wtgcell(c)
+
+                if (flux_avail_rum > 1e12 .or. flux_avail_mg > 1e12 .or. isnan(flux_avail_mg) .or. isnan(flux_avail_rum)) then
```

Comment:
Various hard-coded constants (here and elsewhere) should be made double-precision - e.g., `1.e12_r8`